src/target/llvm/codegen_amdgpu.cc

```diff
@@ -228,23 +233,50 @@ inline int DetectROCMApiVersion() {
   return 305;
 }
 
-runtime::Module BuildAMDGPU(IRModule mod, std::string target) {
+static void UpdateTargetConfig(const String& key, const String& value,
```

Comment:
This is more like "canonicalize" instead of "update". What do you think?

Reply:
I feel like the work "canonicalize" is too strong, because it doesn't really have a canonical form. Maybe "update" is appropriate here, because what it does is to just update the key-value in the config. What do you think?

Reply:
Hmmm...the reason I feel `UpdateTarget` is weird is that it's not clear about what is being updated since we didn't pass new values as argument. For exmple, `dict.update(key, new_value)`. Maybe `ProcessTarget` is better? People will at least know that we are doing something internally with this target.

Reply:
`ProcessTarget` sounds a little bit vague, because it doesn't mention in which way we process it...

We are dealing with a dictionary `Map<String, Object>`, which I call "target config", because it is a configuration used to generate a target, not the target class itself. This method aims to update one of its key-value entry. So what about `UpdateTargetConfigKeyValueEntry`?

Reply:
Or `SetKeyValueEntry`?

Reply:
I vote `UpdateTargetConfigKeyValueEntry` in this case.

Reply:
Thanks! Updated!